app/src/main/java/io/github/jbellis/brokk/ContextManager.java

```diff
@@ -2094,8 +2099,9 @@ private void switchToSession(UUID sessionId) {
         ContextHistory loadedCh = sessionManager.loadHistory(currentSessionId, this);
 
         if (loadedCh == null) {
-            contextHistory = new ContextHistory(new Context(this, "Welcome to session: " + sessionName));
-            sessionManager.saveHistory(contextHistory, currentSessionId);
+            io.toolError("Error while loading history for session '%s'.".formatted(sessionName));
+            createOrReuseSession(DEFAULT_SESSION_NAME);
+            return;
```

Comment:
Error message and new session when switching to a session with unreadable history.